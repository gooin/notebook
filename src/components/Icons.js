import React from 'react';
import IosCar from 'react-ionicons/lib/IosCar';
import IosCafe from 'react-ionicons/lib/IosCafe';
import IosPlaneOutline from 'react-ionicons/lib/IosPlaneOutline';
import IosPizzaOutline from 'react-ionicons/lib/IosPizzaOutline';
import IosCash from 'react-ionicons/lib/IosCash';


const Icons = ({iconName},{color="#6cf"}) => {
    switch (iconName) {
        case "car":
            return <IosCar  fontSize="30px" color={color} />;
        case "cafe":
            return <IosCafe fontSize="30px" color={color}/>;
        case "food":
            return <IosPizzaOutline fontSize="30px" color={color}/>;
        case "travel":
            return <IosPlaneOutline fontSize="30px" color={color}/>;
       case "salary":
            return <IosCash fontSize="30px" color={color}/>;
        default:
            return <IosPlaneOutline fontSize="30px" color={color}/>;

    }
}

export default Icons;