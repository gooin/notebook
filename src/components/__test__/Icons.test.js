import React from 'react';
import {shallow} from 'enzyme';
import Icons from "../Icons";
import toJson from 'enzyme-to-json';


const props = {
    iconName: 'salary'
}


describe('test Icons component', () => {
    let wrapper = shallow(<Icons {...props}/>)
    it('should Tabs rendered', () => {
        expect(wrapper.find('IosCash').length).toBe(1)
    })

    it('check snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})