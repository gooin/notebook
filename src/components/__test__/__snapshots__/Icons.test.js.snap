// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Icons component check snapshot 1`] = `
<IosCash
  beat={false}
  color="#6cf"
  fontSize="30px"
  rotate={false}
  shake={false}
  style={Object {}}
/>
`;
